### PR TITLE
Exposing babashka.nrepl.server-functions

### PR DESCRIPTION
**Pull request by Oddsor**
This change exposes babashka.nrepl.server-functions for programmatic use

Solves #1295 

---

Please answer the following questions and leave the below in as part of your PR.

- [x] I have read the [developer documentation](https://github.com/babashka/babashka/blob/master/doc/dev.md).

- [x] This PR corresponds to an [issue with a clear problem statement](https://github.com/babashka/babashka/blob/master/doc/dev.md#start-with-an-issue-before-writing-code).

- [ ] This PR contains a [test](https://github.com/babashka/babashka/blob/master/doc/dev.md#tests) to prevent against future regressions

- [x] I have updated the [CHANGELOG.md](https://github.com/babashka/babashka/blob/master/CHANGELOG.md) file with a description of the addressed issue.

### CODE REVIEW

**Comment by Oddsor** [comment]
Barely gotten started, but figured a draft is good for discussions!

For example, I think solving this by simply exposing functions from the `babashka.nrepl.server` is sufficient, but maybe more parity with https://github.com/nrepl/nrepl/ is desirable 🤔 

Currently I don't know how to build a test for this, but after compiling and launching an nrepl-server:
```
user=> (babashka.nrepl.server/start-server! {:port 1234})
Started nREPL server at 0.0.0.0:1234
{:socket #object[java.net.ServerSocket 0x7c29cb77 "ServerSocket[addr=/0.0.0.0,localport=1234]"], :future #object[clojure.core$future_call$reify__8544 0x8d2d204 {:status :pending, :val nil}]}
```

and then connecting to it using `lein repl :connect 1234` you are greeted with:

```
$ lein repl :connect 1234
Connecting to nREPL at 127.0.0.1:1234
: Could not resolve symbol: nrepl.core/version user 
nil
Error loading namespace; falling back to user
nil
```

File: src/babashka/main.clj [hunk 1]
```diff
@@ -354,6 +354,13 @@ Use bb run --help to show this help output.
        'clojure.tools.cli tools-cli-namespace
        'clojure.java.shell shell-namespace
        'babashka.core bbcore/core-namespace
+       'babashka.nrepl.server (let [ns-sci (sci/create-ns 'babashka.nrepl.server)]
+                                {'start-server! (sci/new-var 'start-server!
+                                                             (fn [opts]
+                                                               (nrepl-server/start-server! @common/ctx opts))
+                                                             {:ns ns-sci})
+                                 'stop-server! (sci/copy-var nrepl-server/stop-server! ns-sci)
+                                 'parse-opt (sci/copy-var nrepl-server/parse-opt ns-sci)})
```
Comment (borkdude) [inline src/babashka/main.clj:363]: Why do we need to expose this?

Comment (borkdude) [reply src/babashka/main.clj:363]: The `parse-opt` I mean

Comment (borkdude) [reply src/babashka/main.clj:363]: The rest looks good, but I prefer to move this to an `babashka.impl` namespace like we do for the other namespaces.

Comment (Oddsor) [reply src/babashka/main.clj]: Agreed, I'm not sure we need to expose parse-opt; at least I see no need for it for my uses 🤷 

I made an attempt at an impl-namespace! https://github.com/babashka/babashka/pull/1297/commits/e27ece8050d1930c1d654d2cf21d110995159ad1

**Comment by Oddsor** [comment]
I've not written a test for this, mostly because I'm not sure if it's necessary? 🤔 
There are separate nrepl-tests, and a "publics-namespace-test" that already covers most of the additions.

I've manually tested and found it working; a minor bug I found is that stopping the server doesn't kill existing connections (just seems to stop listening for new ones), so maybe a separate issue can be filed for that!

**Pull request merged**